src/localsocketcontroller.h

```diff
@@ -45,7 +45,7 @@ class LocalSocketController final : public ControllerImpl {
   void readData();
   void parseCommand(const QByteArray& command);
 
-  void write(const QJsonObject& json);
+  void write(const QJsonObject& json, const QString& expect = QString());
```

Comment:
The last message with an `expect` parameter is the one that takes precedence when driving this timeout mechanism, which should still work when trying to detect a crashed daemon, though it might be a little sloppier.

Reply:
II wonder if this is an issue. Say the next to last reply is the one that hangs? ... I guess if the next to last reply hangs, but the last one doesn't that means the daemon is doing fine and just maybe had a hiccup. Hm. That deserves a comment in code though I think.

Reply:
I have updated the response handling so that we can now handle multiple outstanding responses. It wasn't that hard and it's probably the better way to handle things.